storage/filesystem.go

```diff
@@ -266,7 +266,7 @@ func (f *Filesystem) WriteFiles(meta metadata.Metadata, files map[string][]byte)
 		// If a fsGroup is defined, Chown all files in the timestamped directory.
 		for filename := range files {
 			// Set the uid to -1 which means don't change ownership in Go.
-			if err := os.Chown(filepath.Join(tsDirName, filename), -1, int(*fsGroup)); err != nil {
+			if err := os.Chown(filepath.Join(f.dataPathForVolumeID(meta.VolumeID), tsDirName, filename), -1, int(*fsGroup)); err != nil {
```

Comment:
This is the minimal fix. This PR taught me about a lot of other possible improvements, they will be part of v0.9.0 instead.
This minimal fix matches what we were doing before: https://github.com/cert-manager/csi-lib/pull/65/files#diff-ba1745ccd4960836e72c7cf9c559b61ecceab32eb78f3be37d56d8c777c1d141L270